content/movie-review-pt2.md

```diff
@@ -0,0 +1,520 @@
+
+Deserialization is the process of reconstructing an object from a byte array. To read the data from an existing account, we must deserialized the account data from bytes into readable objects. We can then update this (deserialized) data and serialize the data into the account again.
+
+To serialize and deserialize account data on Solana, we use the `BorshSerialize` and `BorshDeserialize` macros from the `borsh` crate.
+
+```rust
+use borsh::{BorshSerialize, BorshDeserialize};
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct MovieAccountState {
+    pub is_initialized: bool,
+    pub rating: u8,
+    pub description: String,
+    pub title: String
+}
+```
+
+We can implement additional functionality to our struct using the `impl` keyword. You can read more about the `impl` keyword [here](https://doc.rust-lang.org/std/keyword.impl.html).
+
+For our movie reviews, we want the ability to check whether an account has already been initialized. To do this we’ll create an `is_initilized` function that checks the `is_initialized` field on the `MovieAccountState` struct.
+
+Note that `fn in_initialized` is the function we call and takes in `self` as a parameter (which is the `MovieAccountState` struct), and that `self.is_initilized` is referring to the `is_initilized` field on the `MovieAccountState` struct.
+
+```rust
+impl IsInitialized for MovieAccountState {
+    fn is_initialized(&self) -> bool {
+        self.is_initialized
+    }
+}
+```
+
+## Iterate Accounts
+
+Recall that the `AccountInfo` for all accounts required by an instruction are passing through a single `accounts` argument. In order to parse through the accounts and use them within our instruction, we will need to create an [iterator](https://doc.rust-lang.org/stable/std/iter/) with a mutable reference to the `accounts` argument and assign the `AccountInfo` for each account its own variable. We will do this using the `next_account_info` function from the `account_info` module provided by the `solana_program` crate.
+
+Note that `&mut` means a mutable reference the `accounts` argument. You can read more about references in Rust [here](https://doc.rust-lang.org/book/ch04-02-references-and-borrowing.html) and the `mut` keyword [here](https://doc.rust-lang.org/std/keyword.mut.html).
+
+```rust
+// Get Account iterator
+let account_info_iter = &mut accounts.iter();
+
+// Get accounts
+let initializer = next_account_info(account_info_iter)?;
+let pda_account = next_account_info(account_info_iter)?;
+let system_program = next_account_info(account_info_iter)?;
+```
+
+## Create New Account
+
+Before we can update the data field of an account, we have to first create a new account.
+
+To create a new account within our program we must:
+
+1. Calculate the space and rent required for the account
+2. Have an address to assign the new account
+3. Invoke the system program to create the new account
+
+## Space and Rent
```

Comment:
Let's make this a ### three hashtag subsection of the Create New Account section